### ISSUE TITLE
Move dispatcher sink to the beginning of the ROM to help witgen

### ISSUE THREAD

**Issue by Schaeff** [powdr-labs/powdr#474]
With the introduction of the dispatcher in #392, VMs are essentially turned into dynamic block machines. They expose entry points which can be called from the outside, generating each time one block.

This introduces a difficulty for witgen: once we've computed blocks for all the calls we want, how do we fill the rest of the table?

For static block machines (with blocks of static size), one strategy is the repeat an existing block.
This is not ideal for dynamic block machines:
- as the blocks have dynamic sizes, we cannot guarantee that a previous block will have a size which divides the number of remaining rows, which potentially introduces issues at the bottom of the table, as the last block will be incomplete. Not unfixable, but annoying.
- witgen hugely benefits from repeated rows

What we want instead is to end up in an infinite loop.

Before #392, an infinite loop was required in user code. After #392, an infinite loop is available in the dispatcher. In pseudocode:

```
_jump_to_operation;
operation_0::
// body of operation_0
operation_1::
// body of operation_1
_sink::
loop;
```

Therefore, we want witgen to be able to run `_sink` to fill the rest of the table.

We could expose `_sink` like the other operations of this machine, but this would add another lookup for something which is actually not called, so it runs the risk of being optimised away, or just useless prover overhead.

Instead, the idea is to move `_sink` to the beginning of the dispatcher code, so that its `operation_id` is 0.
As a result, witgen can use the same strategy as for block machines which are not called: try to call them with all arguments being 0, which would work here (whatever the arguments, an infinite loop at `operation_id` 0 generates valid blocks)

**Comment by Schaeff** [comment]
@georgwiese can we close this?

**Comment by georgwiese** [comment]
Yeah, I'd say this was solved by #690
